setup.py

```diff
@@ -198,6 +203,21 @@ def download_files(self):
                 url = self.package_url(package[0], package[1], package[2])
                 self.download_file(url, dest)
 
+    def make_request_with_backoff(self, url):
+        for attempt_number in range(MAX_URL_DOWNLOAD_ATTEMPTS):
+            print('Attempting to retrieve remote jar {url}'.format(url=url))
+            try:
+                return urlopen(url)
+            except HTTPError as e:
+                if e.code == 429:
```

Comment:
Can we change this to perform retries also for other error codes?

It seems that Maven is not consistently using `429`, but also other error codes. Recently, we've seen `502` errors returned from the gateway, but these errors appear to be related to client request throttling.

Including other error codes here would make the installation much more resilient. Thanks

Reply:
Weird. You know, we've been running this install via an automated Jenkins build a lot and haven't encountered anything other than a 429. 

I'm inclined to do as you request on this, but I do have a concern on this... Any time we repeatedly pound a server with requests that we're getting error responses for, we risk being blacklisted by the server admins.

Reply:
Fair point. Good to know that you haven't had any issues recently with the 429 fix. Merging this PR as-is would already be a huge improvement over the status quo. :) Thanks